src/Avalonia.Controls/Primitives/ToggleButton.cs

```diff
@@ -32,20 +32,37 @@ public class ToggleButton : Button
         /// <summary>
         /// Defines the <see cref="Checked"/> event.
         /// </summary>
+        [Obsolete("Use IsCheckedChangedEvent instead.")]
         public static readonly RoutedEvent<RoutedEventArgs> CheckedEvent =
-            RoutedEvent.Register<ToggleButton, RoutedEventArgs>(nameof(Checked), RoutingStrategies.Bubble);
+            RoutedEvent.Register<ToggleButton, RoutedEventArgs>(
+                nameof(Checked),
+                RoutingStrategies.Bubble);
 
         /// <summary>
         /// Defines the <see cref="Unchecked"/> event.
         /// </summary>
+        [Obsolete("Use IsCheckedChangedEvent instead.")]
         public static readonly RoutedEvent<RoutedEventArgs> UncheckedEvent =
-            RoutedEvent.Register<ToggleButton, RoutedEventArgs>(nameof(Unchecked), RoutingStrategies.Bubble);
+            RoutedEvent.Register<ToggleButton, RoutedEventArgs>(
+                nameof(Unchecked),
+                RoutingStrategies.Bubble);
 
         /// <summary>
         /// Defines the <see cref="Unchecked"/> event.
         /// </summary>
+        [Obsolete("Use IsCheckedChangedEvent instead.")]
         public static readonly RoutedEvent<RoutedEventArgs> IndeterminateEvent =
-            RoutedEvent.Register<ToggleButton, RoutedEventArgs>(nameof(Indeterminate), RoutingStrategies.Bubble);
+            RoutedEvent.Register<ToggleButton, RoutedEventArgs>(
+                nameof(Indeterminate),
+                RoutingStrategies.Bubble);
+
+        /// <summary>
+        /// Defines the <see cref="IsCheckedChanged"/> event.
+        /// </summary>
+        public static readonly RoutedEvent<RoutedEventArgs> IsCheckedChangedEvent =
```

Comment:
There is an open question whether a new `IsCheckedChangedEventArgs : RoutedEventArgs` class should be added here like WinUI:

https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.togglesplitbuttonischeckedchangedeventargs?view=windows-app-sdk-1.2

WinUI doesn't currently do anything with this but in Avalonia we could add the `IsChecked` property directly to this event args class. That would pass the current state of the property to all subscribers of the event so they wouldn't need to do a separate property read.

Note that any changes here should apply to `ToggleSplitButton` as well. However, the `IsChecked` property is non-nullable there so the event args usage wouldn't be quite as clean (null checks would always be required).